### PR TITLE
[FIX] simplify table loading

### LINKED ISSUE #4894: [BUG] loading table from csv expects an index column
- Follow up on https://github.com/nilearn/nilearn/pull/4807

- loading events from CSV uses the first column as index and leads to unexpected behavior downstream (missing trial_type column warnings leading to all trials labeled as 'dummy', failure...)

Here is an example with plot_events

```python
import pandas as pd
from nilearn.plotting import plot_event, show
import matplotlib.pyplot as plt

onsets = [0.0, 2.0, 3.0]

trial_types = ["foo", "bar", "baz"]

durations = [1, 1, 1]

events = pd.DataFrame(
    {
        "trial_type": trial_types,
        "onset": onsets,
        "duration": durations,
    }
)

tsv_file = "events.tsv"
events.to_csv(tsv_file, sep="\t", index=False)

csv_file = "events.csv"
events.to_csv(csv_file, sep=",", index=False)

fig = plot_event(csv_file, figsize=(15, 5))
fig.suptitle("CSV files no axis reset")

fig = plot_event(csv_file, figsize=(15, 5))
fig.suptitle("CSV files")
plt.axis([0, 5, 0, 5])

fig = plot_event(tsv_file, figsize=(15, 5))
fig.suptitle("TSV files")
plt.axis([0, 5, 0, 5])

fig = plot_event([csv_file, tsv_file], figsize=(15, 5))
fig.suptitle("CSV and TSV files")
plt.axis([0, 5, 0, 5])

show()

# fail when onset is the first column
events = pd.DataFrame(
    {
        "onset": onsets,
        "trial_type": trial_types,
        "duration": durations,
    }
)

csv_file = "events.csv"
events.to_csv(csv_file, sep=",", index=False)
plot_event(csv_file)
```

```
python /home/remi/github/nilearn/nilearn/tmp/events_csv.py
/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/experimental_paradigm.py:131: UserWarning: 'trial_type' column not found in the given events data.
  warnings.warn(
/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/experimental_paradigm.py:131: UserWarning: 'trial_type' column not found in the given events data.
  warnings.warn(
/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/experimental_paradigm.py:131: UserWarning: 'trial_type' column not found in the given events data.
  warnings.warn(
Traceback (most recent call last):
  File "/home/remi/github/nilearn/nilearn/tmp/events_csv.py", line 52, in <module>
    plot_event(csv_file)
  File "/home/remi/github/nilearn/nilearn/nilearn/plotting/matrix_plotting.py", line 559, in plot_event
    event_copy = check_events(event)
                 ^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/experimental_paradigm.py", line 93, in check_events
    events = _check_columns(events)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/remi/github/nilearn/nilearn/nilearn/glm/first_level/experimental_paradigm.py", line 110, in _check_columns
    raise ValueError(
ValueError: The provided events data has no onset column.
```

Image

Image

Image

Image

also not the plot_event issues:  
- axes not scaled to the content
- yticks do not match 'runs' row in figure after changing axis

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes #4894

Changes proposed in this pull request:

- simplify table loading
- testing to see how many things actually break

### CODE REVIEW

File: nilearn/reporting/tests/test_glm_reporter.py [hunk 1]
```diff
@@ -20,9 +20,8 @@
 @pytest.fixture()
 def flm(tmp_path):
     """Generate first level model."""
-    shapes, rk = ((7, 7, 7, 5),), 3
     mask, fmri_data, design_matrices = write_fake_fmri_data_and_design(
-        shapes, rk, file_path=tmp_path
+        shapes=((7, 7, 7, 5),), rk=3, file_path=tmp_path
```
Comment (bthirion) [inline nilearn/reporting/tests/test_glm_reporter.py:24]: looks unrelated ?

Comment (Remi-Gau) [reply nilearn/reporting/tests/test_glm_reporter.py:24]: yeah left over from playing around

File: nilearn/_utils/tests/test_data_gen.py [hunk 2]
```diff
@@ -583,9 +583,9 @@ def test_generate_fake_fmri_error(rng):
 @pytest.mark.parametrize(
     "shapes", [[(2, 3, 5, 7)], [(5, 5, 5, 3), (5, 5, 5, 5)]]
 )
-@pytest.mark.parametrize("rk", [1, 3])
+@pytest.mark.parametrize("rk", [1, 2])
```
Comment (Remi-Gau) [inline nilearn/_utils/tests/test_data_gen.py:586]: revert to 3 and check if parametrization can be done with other ranks

**Comment by Remi-Gau** [comment]
some visual evidence that the bug is fixed:

running the same script from #4894 

## Before

Image

## After 

Figure_4

**Review by man-shu — APPROVED** [review]
LGTM!
